### DISCUSSION TITLE
Flux normalization

### DISCUSSION THREAD

**Discussion by anubhavdas0907** [micom-dev/micom#75] [Q&A] [answered]
Hello Christian,

Thanks for replying to my previous queries.

The results from "grow" workflow and "cooperative_tradeoff" function, return fluxes.

In the discussion, #50 , you had suggested to add the product of flux and abundance to get overall production. If I want to analyze production by each taxa separately, do I need to multiply the flux with the abundance?

Also, while analyzing results from cooperative_tradeoff, do I need to multiply the returned fluxes with taxa abundance?

Regards
Anubhav

**Comment by cdiener** [comment, marked as answer]
> In the discussion, https://github.com/micom-dev/micom/discussions/50 , you had suggested to add the product of flux and abundance to get overall production. If I want to analyze production by each taxa separately, do I need to multiply the flux with the abundance?

No, that is normalized like you would expect for normal FBA. So the units are mmol/(gDW[taxon] * h) so it's already normalized for 1gDW of mass for each taxon and can be directly compared.

> Also, while analyzing results from cooperative_tradeoff, do I need to multiply the returned fluxes with taxa abundance?

Only if you want to calculate overall production rates, otherwise they are already normalized correctly. 

The upcoming 1.0 version of MICOM will also come with new accessors for the solution, so you will be able to things like.

```
sol = com.cooperative_tradeoff(..., fluxes=True)
sol.production_rates
```

directly.

**Reply by anubhavdas0907** [reply]
Thanks, Christian.

Looking forward to the version 1 release.

Just an additional query.

In case of a 3 taxa community with different abundance of each taxa, if taxa A exports a metabolite with flux V_A, which is taken up by taxa B and C with flux of - V_B, and -V_C, respectively, then should the fluxes be adjusted with abundances, if I want to compare the extent of uptake of the metabolite by B and C? 

Regards
Anubhav

**Reply by cdiener** [reply]
If you want to compare which of the two (A and B) imports it faster, then you don't need to normalize. If you want to know which one takes up more as a whole in the community, you would need to scale with the abundance, like comparing abundance(B)*V_B with abundance(C)*V_C.

**Reply by anubhavdas0907** [reply]
Thanks again, Christian.

This is exactly what I wanted to understand.

Regards
Anubhav
